### ISSUE TITLE
Pre-Release builds are not pushed to Myget

### ISSUE THREAD

**Issue by Jonarw** [oxyplot/oxyplot#2014]
Just noticed that the last Pre-Release package that was pushed to Myget is from 2022-12-03. The AppVeyor CI seems to be broken somehow?

Unfortunatly, I don't have much experience with the CI stuff, so I'm not quite sure what could be the problem. The appveyor.yml apparently hasn't changed in 4 years (apart from yesterday).

Could GitHub Packages provide the same functionality? Could it be a good idea to move there, so we everything in one place?

@oxyplot/admins

**Comment by VisualMelon** [comment]
I'd have no objections to moving to github releases

**Comment by Jonarw** [comment]
I vaguely started to get the feeling that we already had this discussion at some point, and I found it in #1681... So that would be the reason why we've not already done that.
So I guess we still have to figure out why the Myget/AppVeyor stuff broke...

**Comment by VisualMelon** [comment]
Ah yeah. I'll be happy to look into the CI when I'm back to my normal routine in a couple of weeks (equally happy for someone else to fix it in my absence)

**Comment by VisualMelon** [comment]
@HavenDV mentioned using MinVer in https://github.com/oxyplot/oxyplot/pull/1912#issuecomment-1659810397

Would there be any objects to adopting this or something similar and pushing pre-releases to NuGet accordingly?

I'd rather not add a project dependency (I think MinVer could be added as a dotnet tool and just run in the CI?).

Not sure why AppVeyor is complaining; logs aren't much help: https://ci.appveyor.com/project/objorke/oxyplot/builds/47765800

**Comment by HavenDV** [comment]
> I'd rather not add a project dependency (I think MinVer could be added as a dotnet tool and just run in the CI?).

I think you can just install the MinVer package on CI as one of the steps before building the project.
`dotnet add package MinVer`
The tool approach seems more complicated to me, because you will need to save the output and set it as the Version property

**Comment by VisualMelon** [comment]
> I think you can just install the MinVer package on CI as one of the steps before building the project.

That would still make it a dependency in the published packages, and one where it is hard to trace where it is introduced. Clearly I need to try and get my head around all this stuff again (been a while since I had to try and fix one)

**Comment by HavenDV** [comment]
No, it will not add dependencies to the final package. This is a development time package, like Source Generators and some other packages, and they will only be used when building and will not affect the end user of the package in any way.
<img width="814" alt="image" src="https://github.com/oxyplot/oxyplot/assets/3002068/7ed07508-fda5-4b0d-88c9-b075450013b2">

**Comment by VisualMelon** [comment]
That's fun: just to be sure, the published package will have no reference to MinVer etc.? I note there are many nuget packages which list a dependency on it: presumably they are configured in a different manner?

**Comment by HavenDV** [comment]
> That's fun: just to be sure, the published package will have no reference to MinVer etc.? I note there are many nuget packages which list a dependency on it: presumably they are configured in a different manner?

Perhaps, at some point, MinVer did not have the DevelopmentDependency property specified, then it was set by default as a regular package, and therefore it was
Now it will definitely be installed by default with a configuration that will not affect the final package:
```
<PackageReference Include="MinVer" Version="5.0.0-alpha.1">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
```
